cmd/config.go

```diff
@@ -47,15 +47,38 @@ func newConfigCommand(client *client.Client) *Command {
 func runShowConfig(cmdCtx *cmdctx.CmdContext) error {
 	ctx := cmdCtx.Command.Context()
 
-	cfg, err := cmdCtx.Client.API().GetConfig(ctx, cmdCtx.AppName)
+	apiClient := cmdCtx.Client.API()
+	appCompact, err := apiClient.GetAppCompact(ctx, cmdCtx.AppName)
 	if err != nil {
-		return err
+		return fmt.Errorf("error getting app: %w", err)
+	}
+
+	switch appCompact.PlatformVersion {
+	case "nomad":
+		serverCfg, err := apiClient.GetConfig(ctx, cmdCtx.AppName)
+		if err != nil {
+			return err
+		}
+		cmdCtx.WriteJSON(serverCfg.Definition)
+	case "machines":
+		appConfig, err := getAppV2ConfigFromReleases(ctx, apiClient, appCompact.Name)
+		if appConfig == nil {
+			appConfig, err = getAppV2ConfigFromMachines(ctx, apiClient, appCompact)
+		}
+		if err != nil {
+			return err
+		}
+
+		definition, err := appConfig.ToDefinition()
```

Comment:
no need to convert the config into a definition, it should marshal fine when called as `cmxCtx.WriteJSON(appConfig)`